x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_nav.tsx

```diff
@@ -45,18 +47,40 @@ import {
   API_LOGS_TITLE,
 } from './constants';
 
+import { EngineLogic } from './';
+import { EngineDetails } from './types';
+
 import './engine_nav.scss';
 
 export const EngineRouter: React.FC = () => {
```

Comment:
I'm somewhat starting to regret putting EngineRouter and EngineNav in the same file; I thought it made sense at the time because they shared so much logic (e.g. capability checks, path constants) but now I'm wondering if it makes the entire file + test file too unwieldy to read. Do you have strong thoughts here? I'm happy to split these up into two separate files if you do!

Reply:
I'm not a fan of having them in the same file. It's up to you whether you want to change it or not.

Reply:
I think it's not just about verbosity, but I think it's the last place I'd look to find routing logic. It's a little bit off organizationally I think.

Reply:
my thought that nav was related to routing, but yeah I think you're right. I'll split them out!

Reply:
https://github.com/elastic/kibana/pull/83138/commits/2556da78070f9bf28448ddea57f7aa78fa7a2971